src/NClient.Tools/NClient.DotNetTool/Program.cs

```diff
@@ -51,33 +52,74 @@ public static async Task<int> Main(string[] args)
+                    }
+
+                    var clientVerb = typeof(ClientOptions).GetCustomAttribute<VerbAttribute>()?.Name
+                        ?? throw new InvalidOperationException("The attribute with the verb was not found for the facade.");
+                    if (args.First() == clientVerb)
+                    {
+                        var clientParserResult = parser.ParseArguments<ClientOptions.GenerationOptions, int>(args.Skip(1));
+                        return clientParserResult.MapResult(
+                            _ => Task.FromResult(0),
+                            clientErrors => HandleErrors(clientParserResult, clientErrors, showGreeting: false));
+                    }
+
+                    throw new NotSupportedException("Invalid arguments.");
                 });
         }
 
-        private static Task<int> HandleFacadeGenerationOptions(FacadeOptions.GenerationOptions generationOptions)
+        private static Task<int> HandleFacadeGenerationOptions(CommonGenerationOptions generationOptions)
         {
             _serviceProvider = BuildServiceProvider(generationOptions.LogLevel); 
             _logger = _serviceProvider.GetRequiredService<ILogger<Program>>(); 
             return RunFacadeGenerationAsync(generationOptions);
         }
+        
+        private static async Task<int> RunFacadeGenerationAsync(CommonGenerationOptions generationOptions)
+        {
+            try
+            {
+                var specification = await _serviceProvider.GetRequiredService<ILoaderFactory>().Create(generationOptions).Load();
+                var result = await _serviceProvider.GetRequiredService<IFacadeGenerator>().GenerateAsync(generationOptions, specification);
+                await _serviceProvider.GetRequiredService<ISaver>().SaveAsync(result, generationOptions.OutputPath);
+                _logger.LogDone("Generations is over! Please, see {OutputPath} for result!", generationOptions.OutputPath);
+                return 0;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Generation error {Message}", e.Message);
+                return -1;
+            }
+        }
 
-        private static Task<int> HandleErrors(ParserResult<object> parserResult, IEnumerable<Error> errors, bool showGreeting = false)
+        private static Task<int> HandleErrors<T>(ParserResult<T> parserResult, IEnumerable<Error> errors, bool showGreeting = false)
         {
             _serviceProvider = BuildServiceProvider(LogLevel.Trace);
             _logger = _serviceProvider.GetRequiredService<ILogger<Program>>();
                     
-            if (errors.Any(x => x.Tag is ErrorType.HelpVerbRequestedError or ErrorType.NoVerbSelectedError))
+            if (errors.Any(x => x.Tag is ErrorType.HelpVerbRequestedError or ErrorType.HelpRequestedError or ErrorType.NoVerbSelectedError))
```

Comment:
Recommendation: extrude any long-read complex conditions into special scholastic variables for minimize cognitive pressure.

Reply:
I agree, I will do it

Reply:
Done